aiutils/data/data.py

```diff
@@ -0,0 +1,84 @@
+"""Utility functions for making batch generation easier."""
+import numpy as np
+from multiprocessing import Process, Queue
+
+
+def sequential(batch_size, num_samples, num_epochs=1):
+    """Generate sequence indices.
+    """
+    for epoch in range(num_epochs):
+        indices = np.arange(num_samples)
+        for i in range(0, num_samples - batch_size + 1, batch_size):
+            slice_ = slice(i, i+batch_size)
+            yield indices[slice_]
+
+
+def random(batch_size, num_samples, num_epochs=1):
```

Comment:
This could be useful for small datasets where each epoch finishes in a matter of minutes but you'd like to train for a week without specifying an arbitrarily large number.